sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/RuleExecutor.scala

```diff
@@ -118,6 +118,12 @@ class PlanChangeLogger[TreeType <: TreeNode[_]] extends Logging {
 
 abstract class RuleExecutor[TreeType <: TreeNode[_]] extends Logging {
 
+  /** Name for this rule executor, automatically inferred based on class name. */
+  val name: String = {
```

Comment:
How about to convert it to a method with `protected` to allow override in sub-classes for better names.

Reply:
Make sense, let me update it now.